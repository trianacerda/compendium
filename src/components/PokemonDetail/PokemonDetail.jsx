import PropTypes from 'prop-types';

export default function PokemonDetail({ pokemon }) {
  const secondType = pokemon.type_2 !== 'NA';
  const secondAbility = pokemon.ability_2 !== 'NA';

  return (
    <figure aria-label="pokemon">
      <img
        src={pokemon.url_image}
        alt={`${pokemon.pokemon}`}
        style={{
          backgroundColor: 'black',
          borderBlockColor: 'white',
          border: 'dotted',
          margin: '10px',
          height: '200px',
          width: '200px',
        }}
      />
      <div className="pokemon-detail">
        <a
          href={pokemon.pokedex}
          style={{
            border: 'solid',
            color: 'lime',
            justifyContent: 'center',
            displayFlex: 'flex-wrap',
            textAlign: 'center',
          }}
        >
          {pokemon.pokemon}
        </a>
        <p>
          Abilities:
          {pokemon.ability_1}
          {secondAbility && `/${pokemon.ability_2}`}
        </p>
        <p></p>
        <p>Weight: {pokemon.weight}</p>
        <p>Shape: {pokemon.shape}</p>
        <p>
          Types:
          {pokemon.type_1}
          {secondType && `/${pokemon.type_2}`}
        </p>
        <p>Speed:{pokemon.speed}</p>
      </div>
    </figure>
  );
}

PokemonDetail.propTypes = {
  pokemon: PropTypes.shape({
    id: PropTypes.number.isRequired,
    url_image: PropTypes.string.isRequired,
    pokedex: PropTypes.string.isRequired,
    weight: PropTypes.number.isRequired,
    shape: PropTypes.string.isRequired,
    type_1: PropTypes.string.isRequired,
    type_2: PropTypes.string.isRequired,
    speed: PropTypes.number.isRequired,
  }).isRequired,
};
