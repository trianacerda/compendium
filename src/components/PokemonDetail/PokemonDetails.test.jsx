import { screen, render } from '@testing-library/react';
import PokemonDetail from './PokemonDetail';

it('should render a single pokemon', () => {
  render(
    <PokemonDetail
        pokemon={{
          id: 11,
          url_image: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png',
          pokedex: PropTypes.string.isRequired,
          weight: PropTypes.number.isRequired,
          shape: PropTypes.string.isRequired,
          type_1: PropTypes.string.isRequired,
          type_2: PropTypes.string.isRequired,
          speed: PropTypes.number.isRequired,
        }}
  )
})