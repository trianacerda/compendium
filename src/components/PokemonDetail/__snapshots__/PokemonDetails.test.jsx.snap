// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a single pokemon 1`] = `
<p>
  Shape: 
  egg
</p>
`;
